platform/ios/MGLAccountManager.m

```diff
@@ -38,15 +38,15 @@ + (instancetype) sharedInstance {
 }
 
 + (void) setAccessToken:(NSString *) accessToken {
-    [[MGLAccountManager sharedInstance] setAccessToken:accessToken];
+    [[MGLAccountManager sharedManager] setAccessToken:accessToken];
 
     // Update MGLMapboxEvents
     // NOTE: This is (likely) the initial setup of MGLMapboxEvents
-    [MGLMapboxEvents setToken:accessToken];
+    [MGLMapboxEvents sharedManager];
```

Comment:
No, the actual object is held by [a static global](https://github.com/mapbox/mapbox-gl-native/blob/1ec5-metrics-opt-way-out/platform/ios/MGLAccountManager.m#L16), so it will never go away for the lifetime of the process.